solutions/azure-hub-spoke-connected-group/bicep/modules/avnm.bicep

```diff
@@ -0,0 +1,315 @@
+  priority: 100
+  protocol: 'TCP'
+  sourcePortRanges: ['0-65535']
+  sources: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]  
+  }      
+}
+
+@description('This rule to deny inbound TCP/UDP rules')
+resource rule2 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules@2022-05-01' = {
+name: 'r-tcp-udp-${location}'
+kind: 'Custom'
+parent: rulesCollection
+properties: {
+  access: 'Deny'
+  description: 'Deny all TCP/UDP rules'
+  destinationPortRanges: ['11','135','162','593','2049']
+  destinations: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]
+  direction: 'Inbound'
+  priority: 101
+  protocol: 'TCP,UDP'
+  sourcePortRanges: ['0-65535']
+  sources: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]  
+  }      
+}
+
+@description('This rule to deny inbound UDP rules')
+resource rule3 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/rules@2022-05-01' = {
+name: 'r-udp-${location}'
+kind: 'Custom'
+parent: rulesCollection
+properties: {
+  access: 'Deny'
+  description: 'Deny all UDP rules'
+  destinationPortRanges: ['69','11211']
+  destinations: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]
+  direction: 'Inbound'
+  priority: 102
+  protocol: 'UDP'
+  sourcePortRanges: ['0-65535']
+  sources: [
+          {
+            addressPrefix: '*'
+            addressPrefixType: 'IPPrefix'
+          }
+      ]  
+  }      
+}
+
+//
+// In order to deploy a Connectivity or Security configruation, the /commit endpoint must be called or a Deployment created in the Portal. 
+// This DeploymentScript resource executes a PowerShell script which calls the /commit endpoint and monitors the status of the deployment.
+//
+module deploymentScriptConnectivityConfigs './avnmDeploymentScript.bicep' = {
+  name: 'ds-${location}-connectivityconfigs'
+  dependsOn: [
+    roleAssignment
+  ]
+  params: {
+    location: location
+    userAssignedIdentityId: userAssignedIdentity.id
+    configurationIds: '${connectivityConfigurationProd.id},${connectivityConfigurationNonProd.id}' // each configuration separated by a semicolon
+    configType: 'Connectivity'
+    networkManagerName: networkManager.name
+    deploymentScriptName: 'ds-${location}-connectivityconfigs'
+  }
+}
+
+module deploymentScriptSecurityConfigs './avnmDeploymentScript.bicep' = {
+  name: 'ds-${location}-securityconfigs'
+  dependsOn: [
+    roleAssignment
+  ]
+  params: {
+    location: location
+    userAssignedIdentityId: userAssignedIdentity.id
+    configurationIds: securityConfig.id // each configuration separated by a semicolon
+    configType: 'SecurityAdmin'
+    networkManagerName: networkManager.name
+    deploymentScriptName: 'ds-${location}-securityconfigs'
+  }
+}
```

Comment:
Unless I'm missing something, the script as written, doesn't appear to be idempotent.  If we're telling an IaC story, we need to be clear if something in there will break idempotency.  The script appears to blindly run without first checking if something needs to change.  MAYBE the REST endpoint itself is idempotent and that's then all good.  Just calling it out here in case it's not or you could even add a `@description()` on here to let users know that this script needs to be idempotent to support incremental updates with these templates, and this example is.

Reply:
Good point, I'll need to test. I assumed that the PG expects most customers to create deployments/commits using the Portal and included this step to make the solution complete...

Reply:
We don't assume that, matter of fact, we assume this won't be a portal-driven experience.  IaC needs to be idempotent to hold up to its inherit promiss.  Go ahead and make it idempotent or you'll need to call out if this isn't as a work item for a learner to do if taking this towards production.

Reply:
Confirmed that re-running the unchanged template does not result in a re-commit of the same config:

```pwsh
PS C:\Users> Invoke-AzRestMethod -Method POST -uri "https://management.azure.com/subscriptions/.../resourceGroups/rg-avnmmodule/providers/Microsoft.Network/networkManagers/avnm-eastus/listDeploymentStatus?api-version=2022-07-01" -payload @'
>> {
>>   "regions": [
>>     "eastus"
>>   ],
>>   "deploymentTypes": [
>>     "SecurityAdmin"
>>   ]
>> }
>> '@

Headers    : {[Cache-Control, System.String[]], [Pragma, System.String[]], [x-ms-ratelimit-remaining-subscription-writes, System.String[]], [Strict-Transport-Security,
             System.String[]]…}
Version    : 1.1
StatusCode : 200
Method     : POST
Content    : {"skipToken":"","value":[{"deploymentType":"SecurityAdmin","configurationIds":["/subscriptions/24730882-456b-42df-a6f8-8590ca6e4e37/resourceGroups/rg-avnmmod
             ule/providers/Microsoft.Network/networkManagers/avnm-eastus/securityAdminConfigurations/sg-eastus"],"region":"eastus","deploymentStatus":"Deployed","errorMes
             sage":"","commitTime":"2023-01-13T15:44:09.0000000Z"}]}
```